### ISSUE TITLE
SqlBulkCopy: Add RowsCopied Property

### ISSUE THREAD

**Issue by fowl2** [dotnet/SqlClient#136]
It is is difficult to find out how many rows `SqlBulkCopy.WriteToServer[Async]` has copied. This is a problem when using streaming as `DbDataReader` doesn't provide this information either.

### Describe the solution you'd like
```
class SqlBulkCopy
{
...
  public long RowsCopied { get; }
...
}
```

It would be cleaner to change the return type of `WriteToServer[Async]` to `long`, but I think this would technically be a breaking change? This would also have the advantage of not having to worry about multiple calls or reads before writing.

### Describe alternatives you've considered
* setting `NotifyAfter` to 1, subscribing to `SqlRowsCopied` and then storing it somewhere from the event handler. Cons: overhead of calling a delegate, plus the allocation of `EventArgs` per row.
* fully buffer to a `DataTable`. Cons: whole dataset in memory.
* wrapping `IDataReader` and keeping count of the number of times Read() is called. Cons: 
doesn't work for async readers.
* wrapping `DbDataReader`. Cons: lots of boilerplate, can't call protected methods on the source leading to unintended behaviour divergence.
* `SELECT COUNT(*) FROM DESTINATION`. Cons: network round trip, concurrency issues.
* Reflecting on the private `_rowsCopied` field. Cons: reflection, brittle. 

### Additional context
* https://social.msdn.microsoft.com/Forums/sqlserver/en-US/9f83c487-d165-4a9e-a701-d0971de25c4f/c-net-sqlbulkcopy-inserted-row-count?forum=sqlintegrationservices
* https://stackoverflow.com/questions/1188384/sqlbulkcopy-row-count-when-complete

**Comment by cheenamalhotra** [comment]
Hi @fowl2 

Thanks for bringing up the request. We'll investigate more and come back to you!

**Comment by juliusfriedman** [comment]
This is a good suggestion, we currently use reflection to access this field because we don't wan't to have the overhead of an event for every row copied just to get the count.

**Comment by cheenamalhotra** [comment]
@fowl2 @juliusfriedman 

https://github.com/dotnet/SqlClient/issues/404 is same issue we addressed with PR #409 .
The API will be part of next release.
